src/extension/postgres/extension.rs

```diff
@@ -0,0 +1,271 @@
+use crate::{QueryBuilder, QuotedBuilder, SqlWriter};
+
+/// Creates a new "CREATE or DROP EXTENSION" statement for PostgreSQL
+///
+/// # Exampl
+#[derive(Debug, Default, Clone, PartialEq, Eq)]
+pub struct Extension;
+
+impl Extension {
+    /// Creates a new [`ExtensionCreateStatement`]
+    pub fn create() -> ExtensionCreateStatement {
+        ExtensionCreateStatement::new()
+    }
+
+    /// Creates a new [`ExtensionDropStatement`]
+    pub fn drop() -> ExtensionDropStatement {
+        ExtensionDropStatement::new()
+    }
+}
+
+/// Creates a new "CREATE EXTENSION" statement for PostgreSQL
+///
+/// # Synopsis
+///
+/// ```ignore
+/// CREATE EXTENSION [ IF NOT EXISTS ] extension_name
+///     [ WITH ] [ SCHEMA schema_name ]
+///              [ VERSION version ]
+///              [ CASCADE ]
+/// ```
+///
+/// # Example
+///
+/// Creates the "ltree" extension if it doesn't exists.
+///
+/// ```
+/// use sea_query::{extension::postgres::Extension, *};
+///
+/// assert_eq!(
+///     Extension::create()
+///         .name("ltree")
+///         .schema("public")
+///         .version("v0.1.0")
+///         .cascade()
+///         .if_not_exists()
+///         .to_string(PostgresQueryBuilder),
+///     r#"CREATE EXTENSION IF NOT EXISTS ltree WITH SCHEMA public VERSION v0.1.0 CASCADE"#
+/// );
+/// ```
+///
+/// # References
+///
+/// [Refer to the PostgreSQL Documentation][1]
+///
+/// [1]: https://www.postgresql.org/docs/current/sql-createextension.html
+#[derive(Debug, Default, Clone, PartialEq, Eq)]
+pub struct ExtensionCreateStatement {
+    pub(crate) name: String,
+    pub(crate) schema: Option<String>,
+    pub(crate) version: Option<String>,
+
+    /// Conditional to execute query based on existance of the extension.
+    pub(crate) if_not_exists: bool,
+
+    /// Determines the presence of the `RESTRICT` statement
+    pub(crate) cascade: bool,
+}
+
+impl ExtensionCreateStatement {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Sets the name of the extension to be created.
+    pub fn name(&mut self, name: &str) -> &mut Self {
+        self.name = name.to_string();
+        self
+    }
+
+    /// Uses "WITH SCHEMA" on Create Extension Statement.
+    pub fn schema(&mut self, schema: &str) -> &mut Self {
+        self.schema = Some(schema.to_string());
+        self
+    }
+
+    /// Uses "VERSION" on Create Extension Statement.
+    pub fn version(&mut self, version: &str) -> &mut Self {
+        self.version = Some(version.to_string());
+        self
+    }
```

Comment:
A nitpick: can we take a generic `T: Into<String>` instead of `&str`?

Reply:
Actually this is a very good one! Thanks!